### ISSUE TITLE
Access to Django request from within the plugin

### ISSUE THREAD

**Issue by db4y** [codingjoe/django-health-check#290]
Hi

I'm writing a custom check and I would need some informations contained in Django's request object in my check. I could not find a way to access the request nor to pass it to the plugin.

Is it possible to access the request from the plugin?

Thank you.

**Comment by skarzi** [comment]
Hi :wave:

I'm afraid it's not possible because [`CheckMixin`](https://github.com/KristianOellegaard/django-health-check/blob/master/health_check/mixins.py#L9) that is subclassed in [`MainView`](https://github.com/KristianOellegaard/django-health-check/blob/master/health_check/views.py#L80) doesn't pass the `request` object down to plugins `run_check` calls.
So the only way is to rewrite this package flow or (personally I dislike this solution) use some middleware to set currently handled `request` to some global variable

**Comment by db4y** [comment]
Thank you for the quick answer @skarzi 
Do you think it would be possible to add this feature to the package? I would be willing to help.

**Comment by codingjoe** [comment]
Yes, it would be 😉  If you want to contribute this feature I'd be happy to review it. Best, Joe

**Comment by codingjoe** [comment]
Hi there 👋, I will close this due to inactivity. Please don't hesitate to reopen this issue should there be sustained interest in the feature. Cheers, Joe
